### ISSUE TITLE
Downloading or Updating any large Mod lags the UI horrendously

### ISSUE THREAD

**Issue by Custodian19** [geode-sdk/geode#1119]
### Geode Issue

- [X] I confirm that this bug is NOT related to a mod but directly to Geode Loader itself.

### Platform

Android64

### Geode Version

v3.8.1

### Expected Behavior

I expected the Progress Bar to advance smoothly, of course maybe with some minor stops depending on the Internet connectivity

https://github.com/user-attachments/assets/5bc94ac8-5ae1-4c34-b6fc-d4689ed58d4b
(This is a clip of a small mod, which doesn't have that issue as much)

### Actual Behavior

The Game got really laggy, it seemed like it wasn't making any progress, and it seemed to crash, but with enough patiene it did manage to finish.

https://github.com/user-attachments/assets/dcd94923-b7de-41c2-b8cb-479a20d81c4b

### Steps to Reproduce

1) Make sure to have bad internet (e.g. use a VPN far away if necessary, i usually have with normal internet 8 MBit/s download speed)
2) Try and download/update a large mod like QOLMod

### Additional Information

The two video clips were just examples of what can happen,but it can occur with other mods too. In my experience, it can happen with any big mod, that takes more than 1.

And this could also be an Android (9) problem/feature (however i'm not sure), as at first it goes smoothly, but once the Update starts really going far, it seems that the Download Thread moves to the UI Thread. (I'm guessing that's how it's done, as that's a simple and reliable way to do it)

Oh, and while i did have mods installed in the recording, it doesn't matter. I even have the same problems on my Tablet, also running Android 9.

**Comment by HenrySck075** [comment]
not a bug since gd also does that on resources load >:]

**Comment by HenrySck075** [comment]
thats called consistency

**Comment by Custodian19** [comment]
> thats called consistency
That wouldn't really make any sense, even if GD does it like that, since mod loaders usually want to improve ones gameplay experience.

**Comment by HenrySck075** [comment]
So I just spend around 10 minutes looking at the source of the relevant code (and reinstalled qolmod) and I don't think there can be anything from geode that will lag the UI thread 
(tasks are already in its own thread, its listener is called in main tho but there's no download listeners that lag the device)

so my only guess is curl

**Comment by Fleeym** [comment]
It's most likely caused by the curl progress callback, it runs some UI stuff there and the callback gets activated a lot.

I think someone made a fix previously for this, either we have a regression or it still runs a bit bad

**Comment by Custodian19** [comment]
Yeah it's mostly fixed nowadays. I can't seem to replicate it anymore, may also be because I have a newer device.
